### ISSUE TITLE
About hybrid memory re-id loss implementation

### ISSUE THREAD

**Issue by uni20181281** [caposerenity/DAPS#3]
Hi, I have question about the implementation in hm.py loss.

![image](https://user-images.githubusercontent.com/56112192/186341255-0d6def41-8664-4dbc-921c-f84518db4d40.png)

Why did you use masked_softmax in the above code?

and also what is the meaning of below code?

nums = torch.zeros(labels.max()+1, 1).float().cuda()
nums.index_add_(0, labels, torch.ones(self.num_samples,1).float().cuda())
mask = (nums>0).float()
sim /= (mask*nums+(1-mask)).clone().expand_as(sim)

I think this code is same as SPCL re-ID code, but there is not much explantation.

Is using masked re-ID loss neccessary?
